packages/esm-appointments-app/src/appointments/scheduled/scheduled-appointments.component.tsx

```diff
@@ -1,124 +1,120 @@
+      setDateType('futureDate');
+    } else {
+      setDateType('today');
+    }
+  }, [date]);
 
-  const filteredAppointments = appointmentServiceType
-    ? appointmentList.filter(({ serviceTypeUuid }) => serviceTypeUuid === appointmentServiceType)
-    : appointmentList;
-  const rowData = appointmentList.map((appointment, index) => {
-    return {
-      id: `${index}`,
-      ...appointment,
-    };
-  });
-  const filteredRow = appointmentServiceType
-    ? rowData.filter((app) => app.serviceTypeUuid === appointmentServiceType)
-    : rowData;
+  // set the default tab on first render
+  useEffect(() => {
+    if (!currentTab) {
+      setCurrentTab(getDefaultTab(scheduledAppointmentPanels));
+    }
+  }, [scheduledAppointmentPanels]);
 
-  const appointmentsBaseTableConfig = {
-    Scheduled: {
-      appointments: filteredAppointments,
-      isLoading,
-      tableHeading: t('scheduled', 'Scheduled'),
-      visits,
-      scheduleType,
-    },
-    CameEarly: {
-      appointments: earlyAppointmentList,
-      isLoading: loading,
-      tableHeading: t('cameEarly', 'Came early'),
-      visits,
-      scheduleType,
-    },
-    Honoured: {
-      appointments: filteredRow,
-      isLoading,
-      tableHeading: t('honored', 'Honored'),
-      visits,
-      scheduleType,
-    },
-    Rescheduled: {
-      appointments: filteredRow,
-      isLoading,
-      tableHeading: t('rescheduled', 'Rescheduled'),
-      visits,
-      scheduleType,
-    },
-    Pending: {
-      appointments: filteredRow,
-      isLoading,
-      tableHeading: isDateInPast ? t('missed', 'Missed') : t('notArrived', 'Not arrived'),
-      visits,
-      scheduleType,
-    },
-  };
+  // set the default tab whenever the date type changes
+  useEffect(() => {
+    setCurrentTab(getDefaultTab(scheduledAppointmentPanels));
+  }, [dateType]);
 
-  const currentConfig = appointmentsBaseTableConfig[scheduleType];
+  const getDefaultTab = (panels) => {
+    return panels?.find(
+      (panel) =>
+        (isDateInPast && panel.meta.showForPastDate) ||
+        (isToday && panel.meta.showForToday) ||
+        (isDateInFuture && panel.meta.showForFutureDate),
+    )?.meta.title;
+  };
 
   return (
     <>
-      {isToday && (
-        <ContentSwitcher className={styles.switcher} size="sm" onChange={({ name }) => setScheduleType(name)}>
-          <Switch name={'Scheduled'} text={t('scheduled', 'Scheduled')} />
-          <Switch name={'Honoured'} text={t('honored', 'Honored')} />
-          <Switch name={'Pending'} text={t('notArrived', 'Not arrived')} />
-          <Switch name={'CameEarly'} text={t('cameEarly', 'Came early')} />
+      {dateType === 'today' && (
+        <ContentSwitcher className={styles.switcher} size="sm" onChange={({ name }) => setCurrentTab(name)}>
+          {scheduledAppointmentPanels.map((panel) => {
+            return panel.meta.showForToday ? (
+              <Switch name={panel.meta.title} text={t(panel.meta.title, panel.meta.title)} />
```

Comment:
The default value for `t()` is always the key. We only need to use the form with the default value if it would be different than the key.

```suggestion
              <Switch name={panel.meta.title} text={t(panel.meta.title} />
```